jax/lax/lax.py

```diff
@@ -66,20 +66,27 @@
 DType = Any
 Shape = Sequence[int]
 
+def _try_broadcast_shapes(shapes):
+  # Replace 1 with 0 to avoid inconclusive comparisons for polymorphic dims:
+  out_shape = onp.max(onp.where(shapes == 1, 0, shapes), axis=0)
+  out_shape = onp.where(onp.all(shapes == 1, 0), 1, out_shape)
+  out_shape = onp.where(onp.any(shapes == 0, 0), 0, out_shape)
```

Comment:
I think you might be able to safely remove this last line `out_shape = onp.where(onp.any(shapes == 0, 0), 0, out_shape)`.

Any dimensions of size 0 are already preserved in `out_shape`, because dimensions of size 1 were mapped to 0 before taking the max.